pkg/migrations/sql/idempotent/base.sql

```diff
@@ -202,6 +201,14 @@ BEGIN
         END IF;
     END IF;
 
+    --Do not move this into the finalize step, because it's cheap to do while the table is empty
+    --but takes a heavyweight blocking lock otherwise.
+    IF  SCHEMA_CATALOG.is_timescaledb_installed()
+            AND SCHEMA_CATALOG.get_default_compression_setting() THEN
+            PERFORM SCHEMA_PROM.set_compression_on_metric_table(NEW.table_name, TRUE);
```

Comment:
Why do we set compression on a lock metric for maintenance? Wouldnt this lead to unintended result if the user chooses to not compress this metric?

Reply:
Compression is turned on when the metric is created if `SCHEMA_CATALOG.get_default_compression_setting()` is true here.

Reply:
What if user chooses to turn off compression for a metric? Then this will again turn on, even if user didn't want that?